components/script/dom/websocket.rs

```diff
@@ -299,17 +299,11 @@ impl WebSocket {
         let chan = global.r().networking_task_source();
         let address = Trusted::new(self, chan.clone());
 
-        let new_buffer_amount = (self.buffered_amount.get() as u64) + data_byte_len;
-        if new_buffer_amount > (u32::max_value() as u64) {
-            self.buffered_amount.set(u32::max_value());
-            self.full.set(true);
-
-            let _ = self.Close(None, None);
-            return Ok(false);
-
-        }
-
-        self.buffered_amount.set(new_buffer_amount as u32);
+        match data_byte_len.checked_add(self.buffered_amount.get()) {
+            None => return Ok(false),
+            Some(new_amount) => self.buffered_amount.set(new_amount)
+        };
+        // self.buffered_amount.set(self.buffered_amount.get() + data_byte_len);
```

Comment:
`return Ok(false)` isn't really right... the same reasoning about closing the connection applies despite the fact that it's much less likely to happen.  Or, since I can't think of any way to actually trigger this at the moment, I guess we could just `panic!()`.

Also, leftover commented-out code.